Lib/idlelib/debugger.py

```diff
@@ -130,80 +154,187 @@ def make_gui(self):
+
+    def enable_buttons(self, buttons=None):
+        for key in self.buttons:
+            if buttons is None or not key in buttons:
+                self.buttondata[key][0]['image'] = self.buttondata[key][2]
+                self.buttondata[key][0]['foreground'] = '#aaaaaa'
+                self.buttondata[key][0]['cursor'] = ''
+                self.buttondata[key][0].bind('<1>', 'break')
+                self.buttondata[key][0].bind('<<context-menu>>', 'break')
+            else:
+                self.buttondata[key][0]['image'] = self.buttondata[key][1]
+                self.buttondata[key][0]['foreground'] = '#000000'
+                self.buttondata[key][0].bind('<1>', self.button_cmds[key])
+                self.buttondata[key][0].bind('<<context-menu>>',
+                                             self.button_cmds[key])
+                self.buttondata[key][0]['cursor'] = self.clickable_cursor
+
+    def stack_selection_changed(self):
+        self.show_vars()
+
+    def stack_doubleclick(self):
+        sel = self.stack.selection()
+        if len(sel) == 1:
+            self.show_source(sel[0])
+
+    def stack_contextmenu(self, event):
+        item = self.stack.identify('item', event.x, event.y)
+        if item is not None and item != -1 and item != '':
+            menu = Menu(self.top, tearoff=0)
+            menu.add_command(label='View Source',
+                             command = lambda: self.show_source(item))
+            menu.tk_popup(event.x_root, event.y_root)
+
+    def show_source(self, item):
+        if item in self.framevars:
+            fname = self.framevars[item][2]
+            lineno = self.framevars[item][3]
+            if fname[:1] + fname[-1:] != "<>" and os.path.exists(fname):
+                self.flist.gotofileline(fname, lineno)
+
+    def show_status(self, msg, error=False):
+        self.status['text'] = msg
+        self.status['foreground'] = '#ff0000' if error else '#006600'
+        self.status['font'] = self.status_error_font if error \
+                            else self.status_normal_font
+
+    def clear_stack(self):
+        self.stack.delete(*self.stack.get_children(''))
+        self.vars.delete(*self.vars.get_children(self.locals))
+        self.vars.delete(*self.vars.get_children(self.globals))
+        self.vars.detach(self.locals)
+        self.vars.detach(self.globals)
+        self.var_values = {}
+
+    def add_stackframe(self, frame, lineno, current=False):
+        func = frame.f_code.co_name
+        if func in ("?", "", None):
+            func = '.'
+        try:
+            selfval = frame.f_locals['self']
+            if selfval.__class__.__name__ == 'str':
+                # we've probably got the string representation of the
+                # object sent from the remote debugger, see if we can
+                # parse it into something useful
+                match = re.match('^<(?:.*)\.([^\.]*) object at 0x[0-9a-f]+>$',
```

Comment:
```suggestion
                match = re.match('^<(?:.*?)\.([^\.]*?) object at 0x[0-9a-f]+>$',
```
I would prefer if we didn't use greedy lookaheads except when matching the hex. It is also quicker this way, taking [38 steps](https://regex101.com/r/aPLtMa/2) rather than [140 steps](https://regex101.com/r/pWMPKA/2)  on the same test string.

Reply:
I suspect this is correct.  See previous comment about testing REs